File: core/src/com/broll/gainea/server/core/cards/impl/play/C_TeleportPlains.java
package com.broll.gainea.server.core.cards.impl.play;

import com.broll.gainea.server.core.cards.TeleportCard;
import com.broll.gainea.server.core.map.AreaType;
import com.broll.gainea.server.core.map.Location;
import com.broll.gainea.server.core.utils.LocationUtils;

import java.util.List;
import java.util.stream.Collectors;

public class C_TeleportPlains extends TeleportCard {
    public C_TeleportPlains() {
        super(43, "Steppentrip", "Bewegt eine Truppe zu einer beliebigen Steppe auf der gleichen Karte");
    }

    @Override
    public List<Location> getTeleportTargets(Location from) {
        return LocationUtils.filterByType(from.getContainer().getExpansion().getAllLocations(), AreaType.PLAINS).collect(Collectors.toList());
    }
}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_TeleportPlains.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.Game
import com.broll.gainea.server.core.cards.TeleportCard
import com.broll.gainea.server.core.map.AreaType.PLAINS
import com.broll.gainea.server.core.map.Location

class C_TeleportPlains : TeleportCard(43, "Steppentrip", "Bewegt eine Truppe zu einer beliebigen Steppe auf der gleichen Karte") {
    override fun getTeleportTargets(from: Location) = from.container.expansion.allAreas.filter { it.type == PLAINS }

    override fun validFor(game: Game) = game.map.allAreas.any { it.type == PLAINS }
}
